spot_driver/spot_driver/spot_alerts.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+from tkinter import messagebox
+from typing import Any, List, Optional
+
+import bdai_ros2_wrappers.process as ros_process
+from bdai_ros2_wrappers.node import Node
+from rclpy.parameter import Parameter
+
+from spot_msgs.msg import (  # type: ignore
+    BatteryStateArray,
+)
+
+
+class SpotAlerts(Node):
+    """Monitors spot states and will generate a pop-up message to alert the user of warnings"""
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__("spot_alerts", **kwargs)
+        # Subscribers #
+        self.battery_states = self.create_subscription(
+            BatteryStateArray, "status/battery_states", self.battery_callback, 1
+        )
+
+        # Parameters #
+        self.declare_parameter("low_battery", False)
```

Comment:
Why do this as a ROS parameter and not just a class variable?

Reply:
I made it a parameter to make it more easily accessible by other nodes if necessary (maybe some processes should automatically stop if spot has low battery?)

Reply:
I guess I would assume they can also just subscribe to the topic?  Feels more "normal".  But maybe this is more the ROS way?

Reply:
ya for a flag like this I would recommend a parameter since other nodes can have a callback that is only triggered when a specific parameter is changed. A subscriber would run the callback every time they get a new message from the battery_states topic and would also create some duplicate code from this alerts node.

Reply:
Parameters work somewhat differently in ROS 2 vs. ROS 1 -- there isn't a globally-accessible parameter server anymore, so it's not simple for one node to retrieve a parameter from another node. I think subscribing to the battery state topic is the "ROS-y" way to communicate this across multiple nodes. Since the purpose of this specific node is to show a pop-up to the user when the battery level gets low, maybe it's not necessary to have this parameter.

Reply:
The parameters live under the domain of whatever node created it now (as opposed to the master node) but can still be accessed by other nodes. It mostly means you have to specify the node name as well as the parameter name instead of in ROS 1 when you just needed the parameter name

Reply:
Expanding on my "not so simple" comment, it's _possible_ to do this but it's also a big pain that requires a fair amount of finicky code (like the async parameter client in [this demo](https://github.com/ros2/demos/blob/master/demo_nodes_cpp/src/parameters/set_and_get_parameters_async.cpp), for example). I'd generally suggest a different approach for communicating this data between nodes.